thunder/core/dtypes.py

```diff
@@ -80,23 +81,30 @@ def is_weak(self):
         return self._is_weak
 
     def shortname(self):
-        return f"{self._shortname}{8 * self._bytes}"
+        return f"{self._shortname}{8 * self._bytes}{f'_{self._variant}' if self._variant else ''}"
 
     # TODO Fix name printing
     def __repr__(self):
-        return f"{self._name}{8 * self._bytes}{'_' if self._is_weak else ''}"
+        return (
+            f"{self._name}{8 * self._bytes}{f'_{self._variant}' if self._variant else ''}{'_' if self._is_weak else ''}"
```

Comment:
Given that `self._variant` is Optional[str], I think we can write this as 
```python
f"{self._name}{8 * self._bytes}{self._variant if self._variant else ''}{'_' if self._is_weak else ''}"
```

Reply:
This line would make from `f8_variant` to `f8variant`. I think for readability it's maybe better to keep the underscore. What do you think?

Reply:
Ah right, my bad. Thanks for clearing this!